### PR TITLE
show errors when token is not authorized

### LINKED ISSUE #243: Show permission denied error for each task
When using an invalid token the user should be informed that the download fails due to a Permission Error 403

### PR DESCRIPTION
**Pull request by zuzuleinen**
closes #243 

depends on https://github.com/benchkram/bob/pull/256

### CODE REVIEW

File: pkg/store-client/client.go [hunk 1]
```diff
@@ -103,6 +104,8 @@ func (c *c) ListArtifacts(ctx context.Context, project string) (ids []string, er
 
 	if res.StatusCode() == http.StatusNotFound {
 		errz.Fatal(usererror.Wrapm(ErrProjectNotFound, "upload to remote repository failed"))
+	} else if res.StatusCode() == http.StatusUnauthorized {
```
Comment (Equanox) [inline pkg/store-client/client.go:107]: Will this be shown in the logs? I guess not

e.g. on downloading...
```
ui/build       failed to download artifact (permission denied)
ui/build       ... doing a build instead ....
```

Comment (zuzuleinen) [reply pkg/store-client/client.go:107]: pull-unauthorized

For current implementation this is the output of an unauthorized pull

**Comment by Equanox** [comment]
The reason for push is unclear. Should be `not authorized`
image

**Comment by Equanox** [comment]
Some thinks are not perfect yet.
* [ ] cancel pull (It does not react on Ctrl+C)
* [ ] cancel uploads (not tested but i guess it's the same)
* [ ] invalid token on pull should be displayed aligned with build output of the task and also color the task name correctly + why are there empty line in between (generate-go-api - bootstrap)? (can be done in another PR as this probably needs some thinking on how we write to the screen from different places in the code)

image

**Comment by zuzuleinen** [comment]
@Equanox https://github.com/benchkram/bob/pull/256 has to be merged before because it contains some fixes on output as well. Having a branch with https://github.com/benchkram/bob/pull/256 and current PR together will show correct output(for empty line issue)

**Review by Equanox — APPROVED** [review]

**Pull request merged**